### DISCUSSION TITLE
1000 event listners got created in this code?

### DISCUSSION THREAD

**Discussion by sqllyw** [alpinejs/alpine#1129] [1. Help] [answered]
Hi,

with following code, does Alpine call addEventListener() 1000 times? not a good practice when the number is getting big?

````
 <div x-data="">   
    <template x-for="i in 1000">
    <span x-text="i" x-on:click="alert(i)"></span>
    </template>
    
  </div>
  
````

**Comment by KevinBatdorf** [comment, marked as answer]
> not a good practice when the number is getting big?

That's correct, it's probably not a good idea to write your code like that. Instead you can do something like this:
```html
 <div x-data="" x-on:click="alert($event.target.textContent)">   
    <template x-for="i in 1000">
    	<span x-text="i"></span>
    </template>
</div>
```
